content/kapp-controller/docs/latest/debugging.md

```diff
@@ -0,0 +1,113 @@
+or as direct messages from kapp-controller (e.g. when an App uses a ServiceAccount that doesn't exist). 
+
+When a failure occurs with an App CR, you can find further details in the App CR's `DESCRIPTION` column by 
+running `kubectl get apps/simple-app -n namespace`:
+
+```
+NAME         DESCRIPTION                                                                                         SINCE-DEPLOY   AGE
+simple-app   Delete failed: Preparing kapp: Getting service account: serviceaccounts "default-ns-sa" not found   3s             56m
+```
+
+In the case above, the error message shown is coming directly from kapp-controller, so all the information for the failure 
+should be presented in the description column. This commonly occurs when references used by kapp-controller 
+(e.g. secrets, configmaps, serviceaccounts) are not found by kapp-controller.
+
+In cases where the error message does not originate from kapp-controller (e.g. a failed fetch event for a git repository), 
+the stderr from the underlying tool (i.e. vendir, imgpkg, kbld, ytt, kapp, and helm) is shown in the App's status. 
+
+In the App status, there is a field called `usefulErrorMessage` that displays the stderr for a failure during App reconciliation.
+
+This `usefulErrorMessage` field can be found by running `kubectl get apps/simple-app -o=jsonpath={.status.usefulErrorMessage} -n namespace`. 
+The kubectl command will return the stderr output from the App status to help you further understand the reason for the App failure.
+
+The `usefulErrorMessage` can be helpful in pointing out where errors occurred from inputs in the App spec and also 
+pinpoint the resource that caused a deployment failure. However, Apps will not surface errors of resources it is 
+deploying to Kubernetes and further debugging of resources deployed by an App may be needed.
+
+### Debugging InstalledPackage CRs
+
+Failures for InstalledPackages can be directly viewed via the InstalledPackage's status via the `usefulErrorMessage` property of
+the status. This `usefulErrorMessage` property comes from an App CR that is created as a result of creating an InstalledPackage. More 
+information on interpretting the error message from `usefulErrorMessage` can be found under the [Debugging App CRs](#debugging-app-crs). 
+The underlying App CR will have the same name as the InstalledPackage that you create.
+
+In addition to understanding the error from the underlying App CR, it can also be useful the inspect the Package definition to 
+see if there are issues for the Package being deployed by the InstalledPackage.
+
+The name of the Package should be a combination of the Package's `publicName` and `version`. You can view the Package details 
+by taking the `.spec.packageRef.publicName` and `.spec.packageRef.version` values from the InstalledPackage and run the following:
+
+```
+$ kubectl describe package/publicName.version
```

Comment:
This may not be true if an installed package has a version constraint instead of a specific version. I am not sure if we are surfacing the chosen version anywhere on the installedpackage, but if we are, we should use that field, and if we aren't, we should add a pr to do so.

Reply:
It might be best to hold off on this for now as it seems our naming convention should be more clearly defined. I can rephrase.